### ISSUE TITLE
Restoring a RDS instance from snapshot fails to apply an existing parameter group

### ISSUE THREAD

**Issue by hsharma96** [terraform-providers/terraform-provider-aws#6705]
#### Terraform Version
v0.11.10
provider.aws v1.50.0

### Affected Resource(s)

* aws_db_instance

### Terraform Configuration Files

```
provider "aws" {
    region = "us-gov-west-1"
}

# Get latest snapshot from production DB
data "aws_db_snapshot" "latest_snapshot" {
    most_recent             = true
    db_instance_identifier  = "non_prod_dev"
}

# Prints out the latest snapshot name
output "latest_snapshot" {
    value = "${data.aws_db_snapshot.latest_snapshot.id}"
}

variable "parameter_group_name" {
  default = "nonprod-postgres10-copy"
}

# Restore DB from the latest snapshot
resource "aws_db_instance" "latest_snapshot" {
    identifier                  = "testing"
    engine                      = "postgres"
    engine_version              = "10.3"
    license_model               = "postgresql-license"
    instance_class              = "db.m4.xlarge"
    storage_type                = "io1"
    iops                        = "5000"

    apply_immediately           = "true"

    username                    = "xxx"
    password                    = "xxx"

    db_subnet_group_name        = "xzy"
    publicly_accessible         = "true"

    parameter_group_name        = "${var.parameter_group_name}"
    option_group_name           = "default:postgres-10"
    vpc_security_group_ids      = ["xyz", "xzy"]

    auto_minor_version_upgrade  = "false"
    snapshot_identifier         = "${data.aws_db_snapshot.latest_snapshot.id}"
    skip_final_snapshot         = "true"

    tags = {
        Application     = "xyz"
        Environment     = "sbx"
        HasReplica      = "False"
        ReadReplica     = "False"
        Purpose         = "Testing"
    }

    timeouts = {
        create = "60m"
        delete = "2h"
    }
}
```

### Expected Behavior

I would expect the custom parameter group to be set to whatever is defined in the variable.

### Actual Behavior

Unable to assign the existing parameter group to the new instance that's being started by terraform.

This is the error: 
```
Error: Error applying plan:

1 error(s) occurred:

* aws_db_instance.latest_snapshot: 1 error(s) occurred:

* aws_db_instance.latest_snapshot: Error creating DB Instance: InvalidParameterValue: You can't restore a DB instance with a custom ParameterGroup.
```

### Steps to Reproduce

1. `terraform plan`
2. `terraform apply`

**Comment by AngryUbuntuNerd** [comment]
possibly a duplicate of https://github.com/terraform-providers/terraform-provider-aws/issues/1510

**Comment by bflad** [comment]
Thanks, @AngryUbuntuNerd! What is interesting here is that the fix in #1510 went out in version 1.34.0 of the Terraform AWS Provider while the original report here lists version 1.50.0. However, it does seem that this issue is currently fixed in AWS GovCloud (US) according to our daily acceptance testing in that partition (`TestAccAWSDBInstance_SnapshotIdentifier_ParameterGroupName` passes). There may have been some updates to the RDS API in between now and then.

Since this issue appears to be fixed in AWS GovCloud (US) using recent versions of the Terraform AWS Provider, I'm going to close this issue, but if there are still problems please do not hesitate to file a new bug report issue and we will take a fresh look. Thanks!
